# Customer Behaviour, Risk and Decision Analytics

## Objective

This project analyses a bank customer dataset to identify patterns linked to customer churn. The aim is to understand which customer behaviours and characteristics may be associated with customers leaving the bank.

The project focuses on using data analysis to support decision-making, while also considering data quality, limitations, and responsible interpretation of results.



In [ ]:
import pandas as pd

df = pd.read_csv("/var/local/Bank Customer Churn Prediction.csv")
df.head()

,customer_id,credit_score,country,gender,age,tenure,balance,products_number,credit_card,active_member,estimated_salary,churn
0,15634602,619,France,Female,42,2,0.00,1,1,1,101348.88,1
1,15647311,608,Spain,Female,41,1,83807.86,1,0,1,112542.58,0
2,15619304,502,France,Female,42,8,159660.80,3,1,0,113931.57,1
3,15701354,699,France,Female,39,1,0.00,2,0,0,93826.63,0
4,15737888,850,Spain,Female,43,2,125510.82,1,1,1,79084.10,0


## Data Overview

This section explores the structure of the dataset, including the number of rows, columns, column names, and data types.

In [ ]:
df.shape

(10000, 12)

In [ ]:
df.columns

Index(['customer_id', 'credit_score', 'country', 'gender', 'age', 'tenure',
       'balance', 'products_number', 'credit_card', 'active_member',
       'estimated_salary', 'churn'],
      dtype='object')

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customer_id       10000 non-null  int64  
 1   credit_score      10000 non-null  int64  
 2   country           10000 non-null  object 
 3   gender            10000 non-null  object 
 4   age               10000 non-null  int64  
 5   tenure            10000 non-null  int64  
 6   balance           10000 non-null  float64
 7   products_number   10000 non-null  int64  
 8   credit_card       10000 non-null  int64  
 9   active_member     10000 non-null  int64  
 10  estimated_salary  10000 non-null  float64
 11  churn             10000 non-null  int64  
dtypes: float64(2), int64(8), object(2)
memory usage: 937.6+ KB


The dataset contains customer-level information, where each row represents one bank customer.
The target variable is "churn", which indicates whether a customer left the bank. A value of 1 means the customer churned, while 0 means the customer stayed.

## Data Quality Check

Before analysing the data, I checked whether there were any missing values.

In [ ]:
df.isnull().sum()

,0
customer_id,0
credit_score,0
country,0
gender,0
age,0
tenure,0
balance,0
products_number,0
credit_card,0
active_member,0


The dataset contains no missing values, as all columns returned 0 when checked.

This means no missing data needed to be removed or filled in. However, the absence of missing values does not guarantee that the dataset is complete in a wider business context. Important factors such as customer satisfaction, service quality or external financial pressures may not be included.

## Churn Distribution

This section looks at how many customers stayed compared to how many churned.

In [ ]:
df["churn"].value_counts()

,count
churn,
0,7963
1,2037


In [ ]:
df["churn"].value_counts(normalize=True)*100

,proportion
churn,
0,79.63
1,20.37


The dataset shows that 7,963 customers stayed, while 2,037 customers churned.

This means approximately 20% of customers left the bank. Understanding the churn rate is important because it provides a baseline for analysing customer retention.

## Age and Churn

This section compares the average age of customers who stayed with customers who churned.

In [ ]:
df.groupby("churn")["age"].mean()

,age
churn,
0,37.408389
1,44.837997


Customers who stayed have an average age of 37, while customers who churned have an average age of 45.

This suggests that older customers may be more likely to churn. However, this does not prove that age causes churn. It shows a pattern that may be worth investigating further.

##Tenure and Churn

This section compares how long customers have been with the bank and whether this appears linked to churn.

In [ ]:
df.groupby("churn")["tenure"].mean()

,tenure
churn,
0,5.033279
1,4.932744


Customers who stayed have an average tenure of 5.03 years, while customers who churned have an average tenure of 4.93 years.

The difference is very small, suggesting that tenure does not appear to be a strong factor influencing churn in this dataset.

##Activity and Churn

This section explores whether active customers are less likely to churn.

In [ ]:
df.groupby("churn")["active_member"].mean()

,active_member
churn,
0,0.554565
1,0.360825


Approximately 55% of customers who stayed were active members, compared with approximately 36% of customers who churned.

This suggests that customer acitivty is strongly associated with churn. Less active customers appear more likely to leave, which could indicate that customer engagement is an improtant factor in retention.

## Account Balance and Churn

This section compares the average account balance of customers who stayed and customers who churned.

In [ ]:
df.groupby("churn")["balance"].mean()

,balance
churn,
0,72745.296779
1,91108.539337


Customers who stayed had an average balance of approximately 72,745, while customers who churned had a higher average balance of approximately 91,108.

This suggests that customers with higher balances may be more likely to churn. This could be important for the bank because losing higher-balance customers may have a greater financial impact.

## Summary of Findings

The main findings from the analysis are:

- Around 20% of customers in the dataset churned.
- Customers who churned were older on average than customers who stayed.
- Tenure was very similar between both groups, suggesting it may not be a strong churn factor.
- Customers who churned were less likely to be active members.
- Customers who churned had a higher average balance than those who stayed.

The strongest patterns appear to be linked to customer activity and account balance.

## Data Reliability and Risk Considerations

Although the dataset contains no missing values, it may not include all factors that influence customer behaviour.

For example, the dataset does not appear to include customer satisfaction, complaints, service quality, product usage history, or external financial pressures. These missing contextual factors may limit how fully churn can be understood.

Relying only on the available data could lead to oversimplified conclusions. In a real banking environment, further analysis using a wider variety of data would be needed before making decisions about customer retention.

This is especially important because decisions based on customer data should be fair, explainable, and responsible.

## Conclusion

This project analysed a bank customer dataset to identify patterns associated with customer churn.

The analysis found that approximately 20% of customers churned. Customers who churned were older on average, less likely to be active members, and had higher average account balances than customers who stayed. Tenure, however, showed very little difference between the two groups, suggesting it may not be a strong factor in this dataset.

The strongest insight from the analysis is that customer engagement appears to be closely linked to churn. Customers who were less active were more likely to leave, which suggests that improving engagement could be an important customer retention strategy.

The analysis also found that customers with higher balances were more likely to churn, which could represent a business risk because these customers may be financially valuable to the bank.

However, these findings should be interpreted carefully. The dataset does not include every possible factor that may influence churn, such as customer satisfaction, service experience, complaints, or competitor offers. Further analysis using a wider variety of data would be needed to understand the reasons behind churn more fully.

Overall, this project demonstrates how data analysis can be used to identify patterns, support decision-making, and highlight areas where further investigation is needed.